pkg/network/aci_network.go

```diff
@@ -215,14 +216,18 @@ func getDNSConfig(ctx context.Context, pod *v1.Pod, kubeDNSIP, clusterDomain str
 		}
 	}
 
-	if len(nameServers) == 0 {
+	if len(servers) == 0 {
 		return nil
 	}
-	nameServers = formDNSNameserversFitsLimits(ctx, nameServers)
+	servers = formDNSNameserversFitsLimits(ctx, servers)
 	domain := formDNSSearchFitsLimits(ctx, searchDomains)
+	nameServers := make([]*string, len(servers))
```

Comment:
Should you make length 0 here since you are using append?

Reply:
Fixed.